main.go

```diff
@@ -118,9 +126,30 @@ func prefetch(url string) (string, error) {
 	contents, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		log.Print("Prefetch failed reading body for: ", url, err)
+	}
+	dname, err := os.MkdirTemp("", "")
+	if err != nil {
+		log.Fatal(err)
```

Comment:
Don't log the error and instead return it:

```suggestion
		return "", fmt.Errorf("failed to create temp dir: %w", err)
```